main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -1629,4 +1637,22 @@ object Namer {
       */
     case object NonMember extends DefKind
   }
+
+  /**
+    * Companion object for [[SharedContext]]
+    */
+  private object SharedContext {
+    /**
+      * Returns a fresh shared context.
+      */
+    def mk(): SharedContext = new SharedContext(new ConcurrentLinkedQueue())
```

Comment:
We use a queue here, but we're not expecting any popping, right?

Reply:
No, but there is no append-only list thing.